docs/source/api/apollo-server.mdx

```diff
@@ -49,11 +48,11 @@ const server = new ApolloServer({
 
 ##### `typeDefs`
 
-`DocumentNode` or `Array<DocumentNode>`
+`string`, `DocumentNode`, or `Array<DocumentNode>`
 </td>
 <td>
 
-Document or documents that represent your server's GraphQL schema, generated by applying the `gql` tag to valid Schema Definition Language (SDL) strings.
+A valid [Schema Definition Language](../schema/schema/#the-schema-definition-language) (SDL) string, document, or documents that represent your server's GraphQL schema. To generate documents, you can apply the `gql` tag (from [`graphql-tag`](https://www.npmjs.com/package/graphql-tag)) to valid SDL strings.
 
 **Required** unless you provide [`schema`](#schema).
```

Comment:
This "required" thing isn't quite true — `gateway` is also an alternative. (And is not documented at all! Note that when documenting `gateway` there's a bit of a tension between "`gateway` is an option that lets you define a dynamic schema that changes over time and also unrelatedly lets you override how execution works" and "`gateway` is the option for integrating with Apollo Gateway". I think it's probably fair to document it as the latter.)

The fact that there are three different approaches to providing a schema (`gateway` for a dynamic schema probably Apollo Gateway, `schema` for a schema you've already constructed, and `typeDefs`/`resolvers` as a shorthand powered by `@graphql-tools/schema`'s `makeExecutableSchema` function), of which you must choose exactly one, perhaps deserved a bit of text before the table.

And I would put those four options in their own section separate from the rest of "schema options". Maybe "specifying a schema" vs... I dunno... "GraphQL options"? Not sure what the best word for "options that control GraphQL parsing, validation, and execution" would be.

Reply:
1. I updated both the `typeDefs` and `resolvers` options to say they are required unless you specify a `schema` or a `gateway`. 
2. Added the `gateway` option (don't know how that slipped through the cracks 😅)
3. I grouped `typeDefs`, `resolvers`, `schema`, and `gateway` all into one section (which I think makes this whole section a lot clearer from the get-go). I like calling this section `Specifying a schema`, too! 

I feel like the difference between `schema`, `gateway` , or `typeDefs`/`resolvers` is pretty clear with grouping this schema-specifying section together.  Let me know what you think!